src/Microsoft.DotNet.Interactive.HttpRequestParser/HttpSyntaxNode.cs

```diff
@@ -21,22 +21,67 @@ internal abstract class HttpSyntaxNode : HttpSyntaxNodeOrToken
     {
     }
 
-    public override TextSpan Span => _span;
+    public override TextSpan FullSpan => _fullSpan;
+
+    public override bool IsSignificant
+    {
+        get
+        {
+            foreach (var child in ChildNodesAndTokens)
+            {
+                if (child.IsSignificant)
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+    }
+
+    public override string Text => SourceText.ToString(Span);
+
+    public override TextSpan Span
+    {
+        get
+        {
+            var firstSignificantNodeOrToken = ChildNodesAndTokens
+                .FirstOrDefault(n => n.IsSignificant);
+
+            var lastSignificantNodeOrToken = ChildNodesAndTokens
+                .LastOrDefault(n => n.IsSignificant);
+
+            var startOfNonTriviaAndComments =
+                firstSignificantNodeOrToken?.Span.Start ??
+                FullSpan.Start;
+
+            var endOfNonTriviaAndComments =
+                lastSignificantNodeOrToken?.Span.End ??
+                FullSpan.End;
```

Comment:
You probably fixed this already, but in the case where a `HttpSyntaxNode` only contains insignificant text (e.g., comments and whitespace token), looks like `Span` would return the same value as `FullSpan` (and include the insignificant text).

We probably need an early out to return a 0-width span that starts and ends at the start position of the first (insignificant) child in this case...

Of course, if we move this computation to the point when nodes are constructed, we may be able to handle this case more easily over there...

Reply:
I've moved span calculations to `GrowSpan`.